feat(audit-trail): add audit trail documentation (#3970)
scaleway/docs-content: identity-and-access-management/audit-trail/concepts.mdx

---
meta:
  title: Audit Trail - Concepts
  description: Learn fundamental concepts of Scaleway's Audit Trail.
content:
  h1: Concepts
  paragraph: Learn fundamental concepts of Scaleway's Audit Trail.
tags: audit-trail log events access
dates:
  validation: 2024-11-26
---

## Audit Trail

Audit Trail is a tool that holds a record of events and changes performed within a Scaleway Organization. Audit Trail allows you to keep track of all actions, whether successful, attempted, or failed, perfomed on your Scaleway resources.

> With Audit Trail you can view [actions](/identity-and-access-management/audit-trail/concepts/#action) taken by any [principal](/identity-and-access-management/iam/concepts/#principal) in any Project and on any of your Scaleway resources. You can also see the API methods used, the outcome of the action performed (status), and the IP adress from which a principal performed an action.

Author: ```suggestion
With Audit Trail you can view actions taken by any [principal](/identity-and-access-management/iam/concepts/#principal) in any Project and on any of your Scaleway resources. You can also see the API methods used, the outcome of the action performed (status), and the IP address from which a principal performed an action.
```

Audit Trail allows you to view the metadata of the API method used, allowing you to investigate incidents and troubleshoot issues.

## Event

> An event is a log of user activity on your Scaleway resources.

Reviewer: It can be an application activity, and probably also activity performed by Scaleway. The definition should be broader than that

> ## Locacion

Author: ```suggestion
## Location
```

> An event is a record of any activity related to your Scaleway resources, encompassing actions by users, applications, or Scaleway itself.

Author: ```suggestion
An event is a record of any activity related to your Scaleway resources, encompassing actions by users, applications, or Scaleway.
```

> ## Location

Reviewer 2: Is that the official term for Projects and Organizations ? Otherwise I would suggest `Scope` because `Location` may be confused with a geographical location.

In the context of Audit Trail, a Project refers to the place where the Scaleway resources being monitored are located. They can be located in different Scaleway Projects, of different Scaleway Organizations.

## Method

A method in the context of Audit Trail is the API call sent to the Audit Trail API whenever a principal takes an action (whether successful, attempted, or failed) on any Scaleway resource.

## Region

> A region is the geographical area in which your resources are located (Paris, Amsterdam). Audit Trail is not available in Poland.

Author: ```suggestion
A region is the geographical area in which your resources are located (Paris, Amsterdam). Audit Trail is not available for the Poland - Warsaw region.
```

> A region is the geographical area in which your resources are located. They are duplicated on all [Availability Zones](/compute/instances/concepts/#availability-zone) of the selected region (Paris, Amsterdam, or Warsaw).

Reviewer: A region is the geographical area in which your resources are located only for regional resources... but Secret Manager is one, and k8s too.

However, we should remove the sentence "they are duplicated on all..." because it is not always true fir regionalized products, and certainly is not for Secret Manager

Audit Trail allows you to select your resources by region. This allows you to better sort through your resources.

> ## Status

Reviewer: Fabien's latest change displays http status codes and not "success/fail" (you can check on staging). Let's discuss this in squad, if we keep it like than then this paragraph has to be changed

A status is the API response returned whenever an action was taken on a resource by a principal. There are two statuses:

- `200`: this means that the API request was executed.

> - `500`: this means that the API request has failed.

Reviewer 3: We can add the `403` status for permission denied

Reviewer 3: I'm not sure we want to document 500 because these are for internal server error, where the error occurs on the Scaleway servers due to a bug or an incident.

> - `500`: this means that the API request has failed.

Author: ```suggestion
- `403`: this means that the permission to access the API was denied.
```
